pkg/common/vclib/datastore.go

```diff
@@ -133,6 +133,41 @@ func (ds *Datastore) ListFirstClassDisks(ctx context.Context) ([]*FirstClassDisk
 	return objs, nil
 }
 
+// GetFirstClassDiskByName gets a specific first class disks (FCD) on this datastore
+func (ds *Datastore) GetFirstClassDiskByName(ctx context.Context, diskName string) (*FirstClassDisk, error) {
+	m := vslm.NewObjectManager(ds.Client())
+
+	oids, err := m.List(ctx, ds.Reference())
+	if err != nil {
+		glog.Errorf("Failed to list disks. Err: %v", err)
+		return nil, err
+	}
+
+	var ids []string
```

Comment:
Could drop this var and loop, just use `oids` as-is in the main loop.

Reply:
Can do... this is just a copy and paste from an existing function. Will get rid of the loop.